src/main/java/com/synopsys/integration/alert/component/settings/PasswordResetService.java

```diff
@@ -0,0 +1,101 @@
+
+@Component
+public class PasswordResetService {
+    private static final int TEMP_PASSWORD_LENGTH = 50;
+    private static final String TEMPLATE_NAME = "password_reset.ftl";
+    private static final String SUBJECT_LINE = "Alert - Password Reset";
+
+    private final AlertProperties alertProperties;
+    private final UserAccessor userAccessor;
+    private final BaseConfigurationAccessor configurationAccessor;
+
+    @Autowired
+    public PasswordResetService(final AlertProperties alertProperties, final UserAccessor userAccessor, final BaseConfigurationAccessor configurationAccessor) {
+        this.alertProperties = alertProperties;
+        this.userAccessor = userAccessor;
+        this.configurationAccessor = configurationAccessor;
+    }
+
+    public void resetPassword(final String username) throws AlertException {
+        final UserModel userModel = userAccessor.getUser(username)
+                                        .orElseThrow(() -> new AlertDatabaseConstraintException("No user exists for the username: " + username));
+        if (StringUtils.isBlank(userModel.getEmailAddress())) {
+            throw new AlertException("No email address configured for user: " + username);
+        }
+        final ConfigurationModel emailConfig = configurationAccessor.getConfigurationByDescriptorNameAndContext(EmailChannel.COMPONENT_NAME, ConfigContextEnum.GLOBAL)
+                                                   .stream()
+                                                   .findFirst()
+                                                   .orElseThrow(() -> new AlertException("No global email configuration found"));
+        final FieldAccessor fieldAccessor = new FieldAccessor(emailConfig.getCopyOfKeyToFieldMap());
+        final EmailProperties emailProperties = new EmailProperties(fieldAccessor);
+
+        final String tempPassword = RandomStringUtils.randomAlphanumeric(TEMP_PASSWORD_LENGTH);
+        final Map<String, Object> templateFields = Map.of(
+            EmailPropertyKeys.TEMPLATE_KEY_SUBJECT_LINE.getPropertyKey(), SUBJECT_LINE,
+            "tempPassword", tempPassword
+        );
+
+        final EmailTarget passwordResetEmail = new EmailTarget(userModel.getEmailAddress(), TEMPLATE_NAME, templateFields, Map.of());
+        handleSendAndUpdateDatabase(passwordResetEmail, emailProperties, username, tempPassword);
+    }
+
+    private void handleSendAndUpdateDatabase(final EmailTarget passwordResetEmail, final EmailProperties emailProperties, final String username, final String tempPassword) throws AlertException {
+        try {
+            final EmailMessagingService emailService = new EmailMessagingService(alertProperties.getAlertTemplatesDir(), emailProperties);
+            emailService.sendEmailMessage(passwordResetEmail);
+            // Only change the password if there isn't an issue with sending the email
+            userAccessor.changeUserPassword(username, tempPassword);
```

Comment:
I think this should be pulled out of this method and just called itself after you send the email.

Reply:
After the catch blocks?